core/services/chainlink/application.go

```diff
@@ -354,7 +355,12 @@ func NewApplication(ctx context.Context, opts ApplicationOpts) (Application, err
 		return nil, err
 	}
 
-	creServices, err := newCREServices(ctx, globalLogger, opts.DS, keyStore, cfg.Capabilities(), cfg.Workflows(), relayChainInterops, opts.CREOpts)
+	billingClient, err := billing.NewWorkflowClient(opts.Config.Billing().URL())
+	if err != nil {
+		globalLogger.Infof("NewApplication: failed to connect to billing client; %w", err)
```

Comment:
I'm not sure this is sufficient. We should not even be _dialing_, on this path, because it delays startup, up to as long as the timeout is set for. It would be OK to launch the dial eagerly, but it should be on another goroutine. This caller goroutine needs to be able to only construct the client without any dialing.

Reply:
Ahh ok. We were going to create a wrapper for it that can reconnect on failures, but that was a later ticket. I can go ahead and create the basic wrapper now to accomplish what you're asking for.

Reply:
Why don't we just drop the explicit opt-in to blocking in the client constructor? Is that actually required by any other callers?

Reply:
https://github.com/smartcontractkit/chainlink-common/blob/main/pkg/billing/workflow_client.go#L315
https://pkg.go.dev/google.golang.org/grpc#WithBlock
It is deprecated anyways